Middleware/GuzzleRequestSpanning.php

```diff
@@ -30,11 +32,39 @@ public function __invoke(callable $handler): callable
             $promise = $handler($request, $options);
             return $promise->then(
                 function (ResponseInterface $response) {
-                    $this->requestSpanning->finish($response->getStatusCode());
-                    $this->tracing->finishActiveSpan();
-                    return $response;
+                    return $this->onFulfilled($response);
+                },
+                function (RequestException $exception) {
+                    $this->onRejected($exception);
+                    throw $exception;
                 }
             );
         };
     }
+
+    private function onFulfilled(ResponseInterface $response): ResponseInterface
+    {
+        $this->requestSpanning->finish($response->getStatusCode());
+        $this->tracing->finishActiveSpan();
+        return $response;
+    }
+
+    private function onRejected(RequestException $exception): void
+    {
+        // TODO: remove this if $this->requestSpanning->finish adds the error tag (to avoid duplicate)
```

Comment:
This TODO is obsolete: `finish()` does not add the error tag. Even if it did: tags are unique and setting tags again overwrite the previous value.

Reply:
I will remove the todo as it’s now part of the finish. 
There’s no guarantee that tags are unique.
For exemple in go tags are not.

Reply:
Ah, I didn't know about these differences in language specific implementations. I just saw this in the PHP specification:
> If there is a pre-existing tag set for key, it is overwritten.

Reply:
I thought the same at first but it looks like it's implementation dependent.
It's the case with jaeger-php though.